content/systems/filecoin_vm/message/_index.md

```diff
@@ -10,41 +10,139 @@ dashboardTests: 0
 - has non-empty `Params` only if `MethodNum` is zero,
 - has non-negative `GasPrice`,
 - has `GasLimit` that is at least equal to the gas consumption associated with the message's serialized bytes,
 - has `GasLimit` that is no greater than the block gas limit network parameter.
 
-When transmitted individually (before inclusion in a block), a message is packaged as
-`SignedMessage`, regardless of signature scheme used. A valid signed message:
 
-- has a total serialized size no greater than `message.MessageMaxSize`.
+```go
+type Message struct {
+	// Version of this message
+	Version int64
+
+	// Address of the receiving actor.
+	To   address.Address
+	// Address of the sending actor.
+	From address.Address
+
+	Nonce uint64
+
+	// Value to transfer from sender's to receiver's balance.
+	Value BigInt
+
+	// GasPrice is a Gas-to-FIL cost
+	GasPrice BigInt
+	// Maximum Gas to be spent on the processing of this message
+	GasLimit int64
+
+	// Optional method to invoke on receiver, zero for a plain value transfer.
+	Method abi.MethodNum
+	//Serialized parameters to the method (if method is non-zero).
+	Params []byte
+}
+```
+
+There should be several functions able to extract information from the `Message struct`, such as the sender and recipient addresses, the value to be transferred, the required funds to execute the message and others as appropriate.
+
+```go
+func (m *Message) Caller() address.Address {
+	return m.From
+}
+
+func (m *Message) Receiver() address.Address {
+	return m.To
+}
+
+func (m *Message) ValueReceived() abi.TokenAmount {
+	return m.Value
+}
+
+func (m *Message) RequiredFunds() BigInt {
+	return BigAdd(
+		m.Value,
+		BigMul(m.GasPrice, NewInt(uint64(m.GasLimit))),
+	)
+}
+```
+
+Given that Transaction Messages should eventually be included in a Block and added to the blockchain, the validity of the message should be checked.
+
+
+```go
+func (m *Message) ValidForBlockInclusion(minGas int64) error {
+	if m.Version != 0 {
+		return xerrors.New("'Version' unsupported")
+	}
+
+	if m.To == address.Undef {
+		return xerrors.New("'To' address cannot be empty")
+	}
+
+	if m.From == address.Undef {
+		return xerrors.New("'From' address cannot be empty")
+	}
+
+	if m.Value.LessThan(big.Zero()) {
+		return xerrors.New("'Value' field cannot be negative")
+	}
+
+	if m.Value.GreaterThan(TotalFilecoinInt) {
+		return xerrors.New("'Value' field cannot be greater than total filecoin supply")
+	}
+
+	if m.GasPrice.LessThan(big.Zero()) {
+		return xerrors.New("'GasPrice' field cannot be negative")
+	}
+
+	if m.GasLimit > build.BlockGasLimit {
+		return xerrors.New("'GasLimit' field cannot be greater than a block's gas limit")
+	}
+
+	// since prices might vary with time, this is technically semantic validation
+	if m.GasLimit < minGas {
+		return xerrors.New("'GasLimit' field cannot be less than the cost of storing a message on chain")
+	}
+
+	return nil
+}
+```
```

Comment:
Is this just duplicating the prose bullet list above? We should not do this. We should do one or the other (and I think the prose is superior).